### DISCUSSION TITLE
Some binding on footer are hidden when type content in input

### DISCUSSION THREAD

**Discussion by Tim-Cao** [Textualize/textual#3256] [Q&A] [answered]
Could you help me investigate why some binding on footer are hidden when type content in input?

https://github.com/Textualize/textual/assets/52661397/ec86245d-a394-4432-b0cb-fed052eb430d

```
    BINDINGS = [
        Binding("ctrl+c", "quit", "Quit"),
        Binding("shift+insert", "paste", "Paste"),
        Binding("ctrl+u", "delete_left_all", "Delete all to the left")
    ]


        yield Header()
        with Horizontal(id='main-content'):
            yield ListView(
                ListItem(Static("Forward Failure PR", classes="nav-item"), id="nav-1"),
                ListItem(Static("Create PR and Forward", classes="nav-item"), id="nav-2"),
                ListItem(Static("Create TF Ticket", classes="nav-item"), id="nav-3"),
                ListItem(Static("Write Comments", classes="nav-item"), id="nav-4"),
                ListItem(Static("Write Description", classes="nav-item"), id="nav-5")
            )
            with ContentSwitcher(initial="nav-1"):
                with VerticalScroll(id="nav-1"):
                    yield Label("Enter the failure pull request number: ")
                    yield Input(id='failure-pull-request-number')
                with VerticalScroll(id="nav-2"):
                    yield Label("Enter the local branch name: ")
                    yield Input(id='local-branch')
                with VerticalScroll(id="nav-3"):
                    yield Label("Enter the case result id: ")
                    yield Input(id='case-result-id')
                with VerticalScroll(id="nav-4"):
                    yield Label("Enter the Jira ticket number: ")
                    yield Input(id='jira-ticket-number-4')
                with VerticalScroll(id="nav-5"):
                    DESCRIPTION_TYPE = [
                        ("Steps to reproduce", "STR"),
                        ("Test Cases", "TC")
                    ]

                    yield Label("Enter the Jira ticket number: ")
                    yield Input(id='jira-ticket-number-5')
        yield Output(highlight=True, markup=True)
        yield Footer()
```

**Comment by davep** [comment]
That's because <kbd>Ctrl</kbd>+<kbd>U</kbd> is [a binding of `Input`](https://textual.textualize.io/widgets/input/#textual.widgets._input.Input.BINDINGS) and all of the bindings of `Input` are marked as `show=False`.

**Reply by Tim-Cao** [reply]
But show=True, doesn't make it visible neither.
Binding("ctrl+u", "delete_left_all", "Delete all to the left", show=True)

**Reply by TomJGooding** [reply]
Works for me?

```python
from textual.app import App, ComposeResult
from textual.widgets import Input, Footer
from textual.binding import Binding


class MyInput(Input):
    BINDINGS = [
        Binding("ctrl+u", "delete_left_all", "Delete all to the left", show=True),
    ]


class ExampleApp(App):
    BINDINGS = [
        Binding("ctrl+c", "quit", "Quit"),
    ]

    def compose(self) -> ComposeResult:
        yield MyInput("Delete me!")
        yield Footer()


if __name__ == "__main__":
    app = ExampleApp()
    app.run()
```

**Reply by davep** [reply]
Without the full context of the code it's impossible to give a concrete answer to this; but based on your initial question it sounds like you're declaring that on the container (or perhaps screen or even app) within which the `Input` fields are? The reason it's not showing is that the binding on `Input` will override the binding declared on its parent container. To override a child's binding you'd need to make it [a priority binding](https://textual.textualize.io/guide/input/#priority-bindings).

More generally though, binding something specific to `Input` as you seem to be doing there isn't really going to work very well, and the binding [would be in place for any other widget that can receive focus](https://textual.textualize.io/guide/input/#input-focus).

**Reply by Tim-Cao** [reply]
```
class ScriptApp(App):
    BINDINGS = [
        Binding("ctrl+u", "delete_left_all", "Delete all to the left", show=True, priority=True)
    ]
    
    
```
This still didn't work.

**Reply by TomJGooding** [reply]
I expect you want `ctrl+u` to call the `delete_left_all` method from `Input`? You wouldn't declare this binding on the app, see my example above where I created a input subclass.

**Reply by Tim-Cao** [reply]
Do you mean the only way to override is to create a custom input widget not the default input widget?

**Reply by davep** [reply]
Tom and I are both pointing out that to bind an Input action you would need to do so on a widget derived from Input.

**Reply by Tim-Cao** [reply]
I see. It makes codes a little bit verbose. Anyway, thanks

**Reply by TomJGooding** [reply, marked as answer]
Just to add to davep's post, here's a simple example which I hope will help explain this:

```python
from textual.app import App, ComposeResult
from textual.widgets import Input, Footer
from textual.binding import Binding


class ExampleApp(App):
    BINDINGS = [
        Binding("ctrl+c", "quit", "Quit"),
        Binding("ctrl+x", "delete_left_all", "Delete all to the left"),
    ]

    def compose(self) -> ComposeResult:
        yield Input("Delete me!")
        yield Footer()


if __name__ == "__main__":
    app = ExampleApp()
    app.run()
```

Try pressing `ctrl+x`. This won't do anything, as there is no such method in the app. If you check the logs in `textual console`:

```
<action> 'delete_left_all' has no target. Could not find methods '_action_delete_left_all' or 'action_delete_left_all'
```
